cachi2/core/package_managers/generic/models.py

```diff
@@ -0,0 +1,37 @@
+from typing import Literal, Optional
+
+from pydantic import BaseModel, field_validator
+
+
+class LockfileMetadata(BaseModel):
+    """Defines format of the metadata section in the lockfile."""
+
+    version: Literal["1.0"]
+
+
+class LockfileArtifact(BaseModel):
+    """Defines format of a single artifact in the lockfile."""
+
+    download_url: str
```

Comment:
Base model definitions should semantically be introduced in a separate commit to separate data types/interfaces from actual logic.